### DISCUSSION TITLE
未开启UoT时,使用shadowsocks+ws/httpupgrade+tls/none等组合,udp流量会直接暴露给审查者

### DISCUSSION THREAD

**Discussion by christarcher** [XTLS/Xray-core#3626] [General]
今天心血来潮,试了下shadowsocks+ws/httpgrade+tls/none组合(不为别的,纯好奇)
我使用v2rayng/v2rayn做的测试,两个客户端都没有提供设置UoT的选项,也没有默认启用
google了下没有找到相关的结果,所以自己测试了一遍

我预想的是,底层使用基于tcp的传输,那么所有连接都应该是tcp的连接才对

实际测试时我直连服务器时测速正常,但是放到cdn上没法连接
后来发现是dns包没法发过去,再用抓包一看,有很多数据是通过udp发送的

![image](https://github.com/user-attachments/assets/96747ead-414a-448e-8353-b8cd6be9668b)
截图中的是对cdn发送的数据,cdn当然不会有响应(104.16.8.8是cloudflare的ip)

![image](https://github.com/user-attachments/assets/91598458-ebda-404d-befe-2074d5022fec)
我直连自己的服务器udp就有响应
观测直连服务器时的第一个udp包,没有任何头部,是全加密的shadowsocks包
也就是说使用shadowsocks时,无论使用什么底层传输,udp的流量都是直接发送的

![image](https://github.com/user-attachments/assets/6feadeb8-4859-4a54-965d-6a51a00038a9)
tcp包均是使用httpupgrade/ws协议的设计发送的数据包,没有任何问题

所以使用shadowsocks+any的传输组合,针对udp流量是无效的,udp流量会直接暴露给审查者
要实现加密,使用vmess即可
要么就自己写配置文件作为配置
要么就直接使用shadowsocks,不要添加没有意义的伪装

**Comment by RPRX** [comment]
How about DO NOT use Shadowsocks?

**Comment by RPRX** [comment]
~~想了下还是 Reopen 吧，方便别人看到~~
